### DISCUSSION TITLE
Add weekly option to scheduler tab

### DISCUSSION THREAD

**Discussion by kwalker99** [borgbase/vorta#812] [Ideas]
Please can you add weekly backups to the scheduler...

**Comment by sudwhiwdh** [comment]
In theory, it should already be possible to set "Every 168 hours" (1 week) for a backup, right?

It's just not that user-friendly. On the other hand, a drop-down menu with multiple options "Every week", "Every n weeks", "Every n months", "On the first day of each month" ... might become too confusing at some point.

Maybe the option "Backup every x hours at y minutes past the hour" could be extended?

Two ideas:

-  "Backup every x days, y hours, z minutes". And "x" can also be zero.
-  Add two additional lines with the following possible settings: "Every n weeks", "Every n months".

For me the question would also be, if the start and therefore reference value of the settings should/must always be the current system time.

I don't know what feels more natural for the user experience.

By the way: The wording "minutes past the hour" sounds a bit complicated here.

**Comment by kwalker99** [comment]
Why not just add another radio button below daily...

Backup Weekly at "time" Day "day dropdown"

**Comment by kwalker99** [comment]
168 hours doesn't work correctly... in theory it should be 11/06/2020 or 06/11/2020 depending on date format...

What happens instead is:
**Next Backup:** 2020-06-08 01:00

**Comment by kwalker99** [comment]
by making this 240 hours, it is now 7 days from today, looks like a bug to me...

**Next Backup:** 2020-06-11 01:00

**Comment by m3nu** [comment]
Currently we support 2 timers: interval (backup every x days/hours) and fixed (backup at this hour every day).

If you enter >24 hours in the interval scheduler, it will get converted to days automatically. So `7*24 hours` would give you weekly backups.

The starting point != current time, which is what you observed above. This is not a bug.

If you are planning to add a weekly option, I'd add it to the fixed scheduler and make the time span a dropdown (daily, weekly, monthly). We use APScheduler internally, but don't want to expose every single option to keep it simple.

**Comment by sudwhiwdh** [comment]
> If you enter >24 hours in the interval scheduler, it will get converted to days automatically. So 7*24 hours would give you weekly backups.

7*24 would be = 168. Then why does it only seem to work with 240 hours? https://github.com/borgbase/vorta/issues/487#issuecomment-639005412

> but don't want to expose every single option to keep it simple.

:+1:

**Comment by stevenvandenbrandenstift** [comment]
It indeed seems to work with some values, 210 becomes about 4 days, so there seems to be a logic error in the conversion.

**Comment by samuel-w** [comment]
Here is my proposed UI design that replaces the daily and hourly options:
![uiProposal](https://user-images.githubusercontent.com/67625312/86838399-bcccf500-c065-11ea-87ae-eb33d509eaf9.png)


The starting at box label would update to have the same date as the next backup label

**Reply by goebbe** [reply]
@samuel-w 
**Edit**: I just realize that this discussion is basically 5 years old and outdated. It came up in the discussion as new, since there was a new comment by a user. This should probably be closed. 
I deleted my comment. Sorry for the noise.

**Comment by m3nu** [comment]
I'm OK with adding the hour/day/week dropdown, but I find the *Starting At* field needlessly complicated. A user who wants backups every hour won't care when they start.

**Comment by sudwhiwdh** [comment]
> A user who wants backups every hour won't care when they start.

What about users who want to make sure that the backup starts at the hour the system was definitely booted?

**Comment by samuel-w** [comment]
There could be an option to run the backup if it was missed, so users don't have to worry about it. APScheduler does have an option for this https://apscheduler.readthedocs.io/en/stable/userguide.html#missed-job-executions-and-coalescing

Also, the starting at field has to include the time, even in hourly backups. For example, I want to set my backups to run every 12 hours, at 9 am/pm so that I have a backup at the start of the day and the end of the day. The selector can either be QDateTimeEdit, or QDateEdit, and QDateEdit doesn't provide enough granularity for this case.

**Comment by tycho-kirchner** [comment]
Is it possible, to set the time, at which the weekly backup shall be performed?

**Reply by goebbe** [reply]
@tycho-kirchner Thanks for your question: 
Short answer: This is not possible with Vorta 0.10.3.  
Currently, there is no option for "Backup weekly", which would probably require a "week-day" and "time" as parameters.  
Since "backup weekly" has not been implemented as a result of this discussion, I would not hold my breath. 

Note that this discussion is about 5 yeas old and largely outdated.

**Reply by tycho-kirchner** [reply]
Thanks. Next time I have a day off, maybe I'll implement it myself (;

**Reply by goebbe** [reply]
@tycho-kirchner 
If Vorta is running in the background, one can trigger a backup manually by executing: 
`vorta --create yourprofilename`

Using this command, you could also use "cron", "crontab" or "systemd-timers" or similar to get a backup every week at a specific day and a specific time. 
However, using cron (as root) can be tricky if you want to run Vorta (a GUI-programm, run by a user).

On Linux Mint e.g. I use the following cron entry to start Vorta at 7h15 every day: 
`15 7 * * * env DISPLAY=:0 env DBUS_SESSION_BUS_ADDRESS=unix:path=/run/user/1000/bus /bin/bash -l -c "if ( ! pidof -x 'vorta' >/dev/null ); then /home/yourusername/.local/bin/vorta &!; fi"`

Explanation: 
env DISPLAY     is required in cron to start users GUI-program
env DBUS_SESSION_BUS_ADRESS    is required to provide access to DBUS 
using "pidof"   check if Vorta is already running
/bin/bash -l -c  "command"  make sure that environment variables are set correctly

**Reply by tycho-kirchner** [reply]
Thanks for this tip. However, a cron-based backup implies that the `Backup schedule` is set to `Manual only`, which is not practical since autopruning and compaction does not run in this case. Being a commandline-monkey myself, I could also simply use plain borg without vorta. However, I find it convenient and honestly programming a backup schedule in Python does not sound very complicated.

**Reply by goebbe** [reply]
Feel free to provide your pull request. :+1:
